dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.dropwizard;
+
+import java.io.File;
+import java.security.cert.CertificateException;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+
+import javax.net.ssl.SSLException;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.thread.ThreadPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.collect.ImmutableMap;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.metric.DropwizardMeterRegistries;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.dropwizard.logging.AccessLogWriterFactory;
+import com.linecorp.armeria.server.dropwizard.logging.CommonAccessLogWriterFactory;
+import com.linecorp.armeria.server.jetty.JettyService;
+
+import io.dropwizard.jersey.setup.JerseyEnvironment;
+import io.dropwizard.jetty.ConnectorFactory;
+import io.dropwizard.jetty.ContextRoutingHandler;
+import io.dropwizard.server.SimpleServerFactory;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.util.Size;
+import io.dropwizard.validation.MinSize;
+
+@JsonTypeName(ArmeriaServerFactory.TYPE)
+public class ArmeriaServerFactory extends SimpleServerFactory {
```

Comment:
Could just implement `ServerFactory` rather than bring in everything from `SimpleServerFactory`, and copy only necessary pieces.

Reply:
Could you add a `TODO:` or `XXX:` about this?